services/src/main/scala/cromwell/pipeline/service/CromwellService.scala

```diff
@@ -0,0 +1,22 @@
+package cromwell.pipeline.service
+
+import cromwell.pipeline.utils.{CromwellConfig, HttpStatusCodes}
+import play.api.libs.json.JsObject
+
+import scala.concurrent.{ExecutionContext, Future}
+
+class CromwellService(httpClient: HttpClient, cromwellConfig: CromwellConfig)(
+  implicit executionContext: ExecutionContext
+) {
+  def getEngineStatus(): Future[Either[Exception, Int]] = {
```

Comment:
```suggestion
  def getEngineStatus(): Future[Either[CromwellIntegrationException, Int]] = {
```
The whole point of using `Future[Either[_, _]]` instead of simple `Future[_]` is to allow us to narrow types of the expected errors. In fact, `Future[Either[Exception, _]]` almost as useless (in terms of type information) as `Future[_]`.